extensions/azdata/src/extension.ts

```diff
@@ -4,13 +4,25 @@
  *--------------------------------------------------------------------------------------------*/
 
 import * as azdataExt from 'azdata-ext';
-import { findAzdata, IAzdataTool, manuallyInstallOrUpgradeAzdata } from './azdata';
+import * as vscode from 'vscode';
+import * as constants from './constants';
+import { findAzdata, IAzdataTool, manuallyInstallOrUpgradeAzdata, promptForEula } from './azdata';
 import * as loc from './localizedConstants';
+import Logger from './common/logger';
 
 let localAzdata: IAzdataTool | undefined = undefined;
+let eulaAccepted: boolean = false;
 
-export async function activate(): Promise<azdataExt.IExtension> {
+export async function activate(context: vscode.ExtensionContext): Promise<azdataExt.IExtension> {
 	localAzdata = await checkForAzdata();
+	eulaAccepted = !!context.globalState.get(constants.acceptEula);
+	if (!eulaAccepted) {
+		// Don't block on this since we want extension to finish activating without requiring user actions.
+		// If EULA has not been accepted then we will check again while executing azdata commands.
+		promptForEula(context.globalState).then(userResponse => {
```

Comment:
this function is async, any reason you can't just await this?

Reply:
The comment above says why we do not await this. The main reason is we want to finish activating the extension and not make it block behind user input. promptForEula requires user input.

Reply:
Then add a .catch too